Assets/MixedRealityToolkit/_Core/Extensions/LayerExtensions.cs

```diff
@@ -74,19 +60,128 @@ public static int Combine(this LayerMask[] layerMaskList)
             return combinedLayerMask;
         }
 
-        public static LayerMask ToMask(int layer)
+        /// <summary>
+        /// Transform layer id to <see cref="LayerMask"/>
+        /// </summary>
+        /// <param name="layerId"></param>
+        /// <returns></returns>
+        public static LayerMask ToMask(int layerId)
         {
-            return 1 << layer;
+            return 1 << layerId;
         }
 
-        private static int GetLayerNumber(ref int cache, string layerName)
+        /// <summary>
+        /// Gets a valid layer layer id using the layer name.
+        /// </summary>
+        /// <param name="cache">The cached layer id.</param>
+        /// <param name="layerName">The name of the layer to look for if the <see cref="cache"/> is unset.</param>
+        /// <returns>The layer id.</returns>
+        public static int GetLayerId(ref int cache, string layerName)
         {
-            if (cache == InvalidLayer)
+            if (cache == InvalidLayerId)
             {
                 cache = LayerMask.NameToLayer(layerName);
             }
 
             return cache;
         }
+
+#if UNITY_EDITOR
+
+        private static UnityEditor.SerializedProperty tagManagerLayers = null;
```

Comment:
Debating if I should put this in a seperate file in an editor folder.

Reply:
Wow. I didn't think about the walls, floors, etc. potentially being a different layer, but that does open the door for some "Spatial Understanding-Like" capabilities even in VR. I personally really like this change and will approve on my side. Obviously I'll be interested to hear Davids thoughts.

Reply:
Well I _did_ add him as a reviewer. lol